tensorflow/contrib/learn/python/learn/learn_io/data_feeder.py

```diff
@@ -363,6 +363,11 @@ def __init__(self,
 
     num_samples = list(self._x.values())[0].shape[
```

Comment:
Sorry for the delay. I think it's a bit harder to read this way. If `x_is_dict` then you have a list, which never has a `value` attr. It's only in the latter case that you need to check for value. So I think the cases are:

* `x_is_dict`
* else
    * shape has a value attr
    * shape doesn't

In general, I think you want to use `is_tensor` from `tensor_utils.py` rather than checking for the attr.